Error cross-compiling for Windows from Ubuntu

### Godot version

4.0

### System information

Ubuntu 20.04.5

### Issue description

Error when cross compiling for Windows from Ubuntu:

Received error:

`Detected mingw version is not using posix threads. Only posix version of mingw is supported. Use "update-alternatives --config <platform>-w64-mingw32-[gcc|g++]" to switch to posix threads.`

After running the command to switch to posix threads tried to compile:

```
$scons platform=windows
scons: Reading SConscript files ...
Auto-detected 4 CPU cores available for build parallelism. Using 4 cores by default. You can override it with the -j argument.
Using MinGW, arch x86_64
Building for platform "windows", architecture "x86_64", target "editor".
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[ 96%] Compiling platform/windows/os_windows.cpp ...
[100%] progress_finish(["progress_finish"], [])
[100%] platform/windows/os_windows.cpp: In member function 'DWRITE_FONT_WEIGHT OS_Windows::_weight_to_dw(int) const':
platform/windows/os_windows.cpp:933:10: error: 'DWRITE_FONT_WEIGHT_SEMI_LIGHT' was not declared in this scope; did you mean 'DWRITE_FONT_WEIGHT_SEMI_BOLD'?
  933 |   return DWRITE_FONT_WEIGHT_SEMI_LIGHT;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          DWRITE_FONT_WEIGHT_SEMI_BOLD
scons: *** [platform/windows/os_windows.windows.editor.x86_64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:05:28.743]
```

### Steps to reproduce

```
sudo apt-get install mingw-w64
scons platform=windows
update-alternatives --config x86_64-w64-mingw32-gcc
update-alternatives --config x86_64-w64-mingw32-g++
scons platform=windows
```



### Minimal reproduction project

This was from a fresh clone on master.

--- a/SConstruct
+++ b/SConstruct
@@ -645,7 +645,7 @@ if selected_platform in platform_list:
             print(
                 "Detected mingw version is not using posix threads. Only posix "
                 "version of mingw is supported. "
-                'Use "update-alternatives --config <platform>-w64-mingw32-[gcc|g++]" '
+                'Use "update-alternatives --config x86_64-w64-mingw32-g++" '
                 "to switch to posix threads."
             )
             Exit(255)

--- a/platform/windows/os_windows.cpp
+++ b/platform/windows/os_windows.cpp
@@ -69,6 +69,11 @@ __declspec(dllexport) int AmdPowerXpressRequestHighPerformance = 1;
 #define WM_POINTERUPDATE 0x0245
 #endif
 
+// Missing in MinGW headers before 8.0.
+#ifndef DWRITE_FONT_WEIGHT_SEMI_LIGHT
+#define DWRITE_FONT_WEIGHT_SEMI_LIGHT (DWRITE_FONT_WEIGHT)350
+#endif
+
 #if defined(__GNUC__)
 // Workaround GCC warning from -Wcast-function-type.
 #define GetProcAddress (void *)GetProcAddress

